src/MongoDB.Driver.Core/Core/Clusters/Cluster.cs

```diff
@@ -222,9 +222,9 @@ private void RapidHeartbeatTimerCallback(object args)
 
         protected abstract void RequestHeartbeat();
 
-        protected void OnDescriptionChanged(ClusterDescription oldDescription, ClusterDescription newDescription)
+        protected void OnDescriptionChanged(ClusterDescription oldDescription, ClusterDescription newDescription, bool shouldClusterDescriptionChangedEventBePublished)
         {
-            if (_descriptionChangedEventHandler != null)
+            if (shouldClusterDescriptionChangedEventBePublished && _descriptionChangedEventHandler != null)
```

Comment:
I think the `Event` should *always* be raised.

It is up to the receiver to decide what to do (or not do).

Reply:
See comment in earlier review on this same topic:

https://github.com/DmitryLukyanov/mongo-csharp-driver/pull/83#discussion_r354090870

Reply:
I'm not sure how it can look like. At least if we pass an event into EventCapturer, then this even will be asserted in the test runner. So, if I remove this new part in the above condition, this test https://github.com/DmitryLukyanov/mongo-csharp-driver/pull/83/files?file-filters%5B%5D=.cs&file-filters%5B%5D=.json#diff-97b7222883d33cefce025d2b22e41bb5R2 will fail

Reply:
I see now what confused me...

We raise this event TWICE in this method, once using our own event mechanism and once using C# events.

Your change is ONLY suppressing the FIRST of the two events, so we are in fact ALWAYS raising the SECOND event, which matches what I expected in my first comment in this conversation.